src/components/shift-schedule/ScheduleCellDialog.tsx

```diff
@@ -0,0 +1,53 @@
+import { RegisterSection } from '@/components/shift-schedule/RegisterSection';
+import { DialogHeader, DialogTitle } from '@/components/ui/Dialog';
+import { useTranslations } from 'next-intl';
+
+type ScheduleCellDialogProps = {
+  tDialog: {
+    day: string;
+    time: string;
+  };
+  members: {
+    name: string;
+  }[];
+};
+
+function ScheduleCellDialog({ tDialog, members }: ScheduleCellDialogProps) {
+  const t = useTranslations(
+    'shiftSchedule.scheduleTable.scheduleCell.scheduleCellDialog',
+  );
+
+  let membersDisplay: React.ReactNode;
```

Comment:
You do not need to declare this variable beforehand, as no matter what the state of `members.length` is, it will receive a value. TS will also understand this is a `JSX.Element`. If you decide to remove this declaration beforehand, you can set the variable as a `const` in the if-statement for both cases (because we only set the variable once, so `const` is prefered over `let`)

Reply:
You may also experiment with making this a conditional inside the return statement. If you think it is not readable, revert back to your solution.